cmake/README.md

```diff
@@ -101,6 +102,11 @@ Use `ONEDPL_PAR_BACKEND` variable before the invocation of `find_package(oneDPL
 - Macro `ONEDPL_USE_TBB_BACKEND` is set to `0` if OpenMP backend is chosen.
 - Macro `ONEDPL_USE_TBB_BACKEND` is set to `0` and `ONEDPL_USE_OPENMP_BACKEND` is set to `0` if serial backend is chosen.
 
+### Using oneDPL package on Windows
+On windows, CMake requires some workarounds to use icx[-cl] and icpx successfully.  A cmake package has been provided 'WindowsIntelLLVM' to apply these required workarounds.
+We recommend using icx or icx-cl, as it requires fewer changes for proper usage.  We also recommend updating to the most recent version of CMake, as they are actively improving support for intel compilers (https://gitlab.kitware.com/cmake/cmake/-/issues/24314).
+To enable the workarounds, please add `find_package(WindowsIntelLLVM)` to your cmake file before you call `project()`.
```

Comment:
Have you tested the following two ways of oneDPL integration through cmake?

- As source (e.g. cloned from GitHub):
```
cmake_minimum_required(VERSION 3.15)
find_package(WindowsIntelLLVM)
project(Foo)
add_executable(foo main.cpp)
add_subdirectory("path/to/oneDPL" build_oneDPL)
target_link_libraries(foo oneDPL)
```
- As an installed package (e.g. a part of [Intel® oneAPI Base Toolkit](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwigvPuqxaf-AhWQBzQIHeyqDD4QFnoECBMQAQ&url=https%3A%2F%2Fwww.intel.com%2Fcontent%2Fwww%2Fus%2Fen%2Fdeveloper%2Ftools%2Foneapi%2Fbase-toolkit.html&usg=AOvVaw1cYyLJpZCyti5Jdeyn5aPP)): 
```
cmake_minimum_required(VERSION 3.15)
find_package(WindowsIntelLLVM)
project(Foo)
add_executable(foo main.cpp)
find_package(oneDPL REQUIRED)
target_link_libraries(foo oneDPL)
```

Reply:
I have tested this for the installed package, and in building oneDPL's tests themselves which should be similar to the first case described.  I will add some testing for an external repo adding oneDPL cloned from github as well.

I will do more testing tomorrow and confirm with the recent changes.

Reply:
I found some issues with the inclusion from another project (first listed below), and resolved them.  In this case, we are too late to automatically include the workarounds, so it is a separate case from the oneDPL tests themselves.  I've added more to the cmake/README.md to guide users to the appropriate usage.

I've tested it now in three ways:  

- As source (e.g. cloned from GitHub):
```
cmake_minimum_required(VERSION 3.15)
list(APPEND CMAKE_PREFIX_PATH /path/to/oneDPL/cmake)
find_package(oneDPLWindowsIntelLLVM)
project(Foo)
add_executable(foo main.cpp)
add_subdirectory("path/to/oneDPL" build_oneDPL)
target_link_libraries(foo oneDPL)
```

- As an installed package (e.g. a part of [Intel® oneAPI Base Toolkit](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwigvPuqxaf-AhWQBzQIHeyqDD4QFnoECBMQAQ&url=https%3A%2F%2Fwww.intel.com%2Fcontent%2Fwww%2Fus%2Fen%2Fdeveloper%2Ftools%2Foneapi%2Fbase-toolkit.html&usg=AOvVaw1cYyLJpZCyti5Jdeyn5aPP)):

```
cmake_minimum_required(VERSION 3.15)
find_package(oneDPLWindowsIntelLLVM)
project(Foo)
add_executable(foo main.cpp)
find_package(oneDPL REQUIRED)
target_link_libraries(foo oneDPL)
```
- Building tests in oneDPL as cloned from GitHub.  This will automatically include the workarounds.

Reply:
I tested these 3 scenarios with VS2019 and VS2022 with a recent commit(cba552328c32ed366bc14b2cf15e92edab624d1f). Everything works as expected.